src/elfpy/markets.py

```diff
@@ -67,67 +95,117 @@ def __init__(
+
         # for each position, specify how to forumulate trade and then execute
-        if user_action.action_type == "open_long":  # buy to open long
-            user_action.direction = "out"  # calcOutGivenIn
-            user_action.token_out = "pt"  # buy unknown PT with known base
-            market_deltas, wallet_deltas = self._open_long(user_action)
-        elif user_action.action_type == "close_long":  # sell to close long
-            user_action.direction = "out"  # calcOutGivenIn
-            user_action.token_out = "base"  # sell known PT for unkonwn base
-            market_deltas, wallet_deltas = self._close_long(user_action)
-        elif user_action.action_type == "open_short":  # sell PT to open short
-            user_action.direction = "out"  # calcOutGivenIn
-            user_action.token_out = "base"  # sell known PT for unknown base
-            market_deltas, wallet_deltas = self._open_short(user_action)
-        elif user_action.action_type == "close_short":  # buy PT to close short
-            user_action.direction = "in"  # calcInGivenOut
-            user_action.token_in = "base"  # buy known PT for unknown base
-            market_deltas, wallet_deltas = self._close_short(user_action)
+        if agent_action.action_type == "open_long":  # buy to open long
+            agent_action.direction = "out"  # calcOutGivenIn
+            agent_action.token_out = "pt"  # sell known base for unknown PT 
+            market_deltas, agent_deltas = self._open_long(agent_action)
+        elif agent_action.action_type == "close_long":  # sell to close long
+            agent_action.direction = "out"  # calcOutGivenIn
+            agent_action.token_out = "base"  # sell known PT for unknown base
+            market_deltas, agent_deltas = self._close_long(agent_action)
+        elif agent_action.action_type == "open_short":  # sell PT to open short
+            agent_action.direction = "in"  # calcOutGivenIn
+            agent_action.token_out = "pt"  # sell known PT for unknown base
+            market_deltas, agent_deltas = self._open_short(agent_action)
+        elif agent_action.action_type == "close_short":  # buy PT to close short
+            agent_action.direction = "in"  # calcOutGivenIn
+            agent_action.token_in = "pt"  # buy known PT for unknown base
+            market_deltas, agent_deltas = self._close_short(agent_action)
+        elif agent_action.action_type == "add_liquidity":
+            # pricing model computes new market deltas
+            # market updates its "liquidity pool" wallet, which stores each trade's mint time and user address
+            # LP tokens are also storeds in user wallet as fungible amounts, for ease of user
+            market_deltas, agent_deltas = self._add_liquidity(agent_action)
+            pass
+        elif agent_action.action_type == "remove_liquidity":
+            # market figures out how much the user has contributed (calcualtes their fee weighting)
```

Comment:
I think it would. make more sense for these comments to live inside the function docstring for the `_remove_liquidity` class.